src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/config/WebClientConfiguration.kt

```diff
@@ -1,71 +1,71 @@
 package uk.gov.justice.digital.hmpps.prisonerfromnomismigration.config
 
-import org.springframework.beans.factory.annotation.Qualifier
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
+import org.springframework.http.client.reactive.ReactorClientHttpConnector
 import org.springframework.security.oauth2.client.AuthorizedClientServiceReactiveOAuth2AuthorizedClientManager
 import org.springframework.security.oauth2.client.ReactiveOAuth2AuthorizedClientManager
 import org.springframework.security.oauth2.client.ReactiveOAuth2AuthorizedClientProviderBuilder
 import org.springframework.security.oauth2.client.ReactiveOAuth2AuthorizedClientService
 import org.springframework.security.oauth2.client.registration.ReactiveClientRegistrationRepository
 import org.springframework.security.oauth2.client.web.reactive.function.client.ServerOAuth2AuthorizedClientExchangeFilterFunction
-import org.springframework.stereotype.Component
 import org.springframework.web.reactive.function.client.WebClient
-import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.health.HealthCheck
+import reactor.netty.http.client.HttpClient
+import java.time.Duration
+import kotlin.apply as kotlinApply
 
 @Configuration
 class WebClientConfiguration(
   @Value("\${api.base.url.nomis}") val nomisApiBaseUri: String,
   @Value("\${api.base.url.oauth}") val oauthApiBaseUri: String,
   @Value("\${api.base.url.mapping}") val mappingApiBaseUri: String,
+  @Value("\${api.health-timeout:2s}") val healthTimeout: Duration,
+  @Value("\${api.timeout:90s}") val timeout: Duration,
 ) {
 
   @Bean
-  fun nomisApiHealthWebClient(): WebClient = WebClient.builder().baseUrl(nomisApiBaseUri).build()
+  fun nomisApiHealthWebClient(builder: WebClient.Builder): WebClient = builder.healthWebClient(nomisApiBaseUri, healthTimeout)
```

Comment:
There was a bug in the past where sharing the single instance of WebClient.Builder meant that it was retaining state between calls to `.build()`. I can't remember how it was presenting - but worth keeping an eye on in case it's still a problem.